Update security docs to be more precise about Ray Client, Dashboard, Jobs (#43858)
ray-project/ray: SECURITY.md

# Security Policy

> If you are deploying Ray, please check out [this documentation page](https://docs.ray.io/en/latest/ray-security/index.html).

Reviewer: ```suggestion
If you are deploying Ray, read [Security](ray-security/index.html).
```

> ## Reporting a Vulnerability

Reviewer: ```suggestion
## Reporting a vulnerability
```

> Please report security issues to `security@anyscale.com`.

Reviewer: ```suggestion
Report security issues to `security@anyscale.com`.
```
